torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -637,11 +637,22 @@ static void fuseListConstructListUnpack(Block* b) {
     for (auto* child_block : it->blocks()) {
       fuseListConstructListUnpack(child_block);
     }
-    if (it->kind() == prim::ListUnpack &&
-        it->input()->node()->kind() == prim::ListConstruct) {
+    if (it->kind() == prim::ListUnpack) {
       for (size_t i = 0; i < it->outputs().size(); i++) {
         auto output = it->outputs().at(i);
-        output->replaceAllUsesWith(it->input()->node()->inputs().at(i));
+        if (it->input()->node()->kind() == prim::ListConstruct) {
+          output->replaceAllUsesWith(it->input()->node()->inputs().at(i));
+        } else if (it->input()->type()->cast<ListType>()) {
```

Comment:
Also wondering is this check enough? 
1. will there be list type input that does not work?
2. will there be other type input (tensor? i'm thinking if it's possible through inferred shape from onnx shape inference) that does work?

Reply:
I'm not sure why would element type of a list have any impact?
I can write a test for tensor type.